contracts/governance/Staking/Staking.sol

```diff
@@ -373,6 +477,46 @@ contract Staking is
         emit StakingWithdrawn(msg.sender, amount, until, receiver, isGovernance);
     }
 
+    /**
+     * @notice Send user' staked tokens to a receiver.
+     * This function is dedicated only for governance vesting withdrawal.
+     *
+     * @param amount The number of tokens to withdraw.
+     * @param until The date until which the tokens were staked.
+     * @param receiver The receiver of the tokens. If not specified, send to the msg.sender
+     * @param vesting The vesting address which act as the caller here.
+     * or just unlocked tokens (false).
+     * */
+    function _withdrawDirect(
+        uint96 amount,
+        uint256 until,
+        address receiver,
+        address vesting
+    ) internal {
+        // @dev it's very unlikely some one will have 1/10**18 SOV staked in Vesting contract
+        if (amount == 1) {
+            return;
+        }
+
+        until = _adjustDateForOrigin(until);
+        _validateWithdrawParams(vesting, amount, until);
+
+        /// @dev Determine the receiver.
+        if (receiver == address(0)) receiver = vesting;
+
+        /// @dev Update the checkpoints.
+        _decreaseDailyStake(until, amount);
+        _decreaseUserStake(vesting, until, amount);
+        if (isVestingContract(vesting)) _decreaseVestingStake(until, amount);
```

Comment:
isVestingContract(vesting) - shouldn't it be be a require statement at the beginning of the function?